column_buffer.go

```diff
@@ -2234,11 +2238,13 @@ func writeRowsFuncOfStruct(t reflect.Type, schema *Schema, path columnPath) writ
 	columns := make([]column, len(fields))
 
 	for i, f := range fields {
+		list := false
 		optional := false
 		columnPath := path.append(f.Name)
 		forEachStructTagOption(f, func(_ reflect.Type, option, _ string) {
 			switch option {
 			case "list":
+				list = true
```

Comment:
It's getting tricky to track the logic with boolean values passed through recursive calls, the combination of `optional`, `list`, and `optionalList` should be documented or it'll be really difficult to maintain this code.

Reply:
Thank you Achille for the review.
I'll document the logic and try to make it a bit better readable. Do you think it could help to capture this in an enum like type?

Reply:
> Do you think it could help to capture this in an enum like type?

Try it! Happy to take this approach if it's more readable.